File: java/core/src/org/daisy/urakawa/media/ILocated.java
package org.daisy.urakawa.media;

import java.net.URI;
import java.net.URISyntaxException;

import org.daisy.urakawa.exception.MethodParameterIsEmptyStringException;
import org.daisy.urakawa.exception.MethodParameterIsNullException;

/**
 * Specifies the location of the data resource for a media object.
 */
public interface ILocated
{
    /**
     * @return the location of the media
     */
    String getSrc();

    /**
     * @param newSrc
     * 
     * @tagvalue Events "SrcChanged"
     * @throws MethodParameterIsNullException NULL method parameters are
     *         forbidden
     * @throws MethodParameterIsEmptyStringException Empty string '' method
     *         parameters are forbidden
     */
    void setSrc(String newSrc) throws MethodParameterIsNullException,
            MethodParameterIsEmptyStringException;

    /**
     * Gets the URI of this location based on getPresentation().getRootURI()
     * 
     * @return a URI
     * @throws URISyntaxException when the value returned by getSrc() is not a
     *         well-formed URI
     */
    public URI getURI() throws URISyntaxException;
}

File: csharp/core/media/ILocated.cs
using System;

namespace urakawa.media
{
	/// <summary>
	/// This interface associates a media object with its source location
	/// </summary>
	public interface ILocated
	{

		/// <summary>
		/// Get the src location of the external media
		/// </summary>
		/// <returns>The src location</returns>
		string getSrc();

		/// <summary>
		/// Set the external media's src location.
		/// </summary>
		/// <param name="newSrc">The new src location</param>
		void setSrc(string newSrc);

		/// <summary>
		/// Gets the <see cref="Uri"/> of the <see cref="ExternalMedia"/> 
		/// - uses <c>getMediaFactory().getPresentation().getRootUri()</c> as base <see cref="Uri"/>
		/// </summary>
		/// <returns>The <see cref="Uri"/></returns>
		/// <exception cref="exception.InvalidUriException">
		/// Thrown when the value returned by <see cref="getSrc"/> is not a well-formed <see cref="Uri"/>
		/// </exception>
		Uri getUri();
	}
}
